### DISCUSSION TITLE
Accessing current session from the client side

### DISCUSSION THREAD

**Discussion by john-kuo** [kinngh/shopify-node-express-mongodb-app#119] [Q&A] [answered]
hello @kinngh, came across another problem.

If I would like to access user's current session information similar to the shopify turorial [here](https://shopify.dev/docs/apps/getting-started/build-qr-code-app?framework=remix)

its' line 20 in their **app._index.jsx**

The example is with remix
Where it gets the session info with this code snippet 
`const { session } = await authenticate.admin(request);
  const { shop } = session;` 
  
  Is there something equivalent I can do on the client side with your project?
  
  Thanks for your help

**Comment by kinngh** [comment, marked as answer]
1. You shouldn't be accessing `session` client side - it's not safe since you're exposing user's online session tokens and your app's static offline tokens.
2. To get tokens on the server, on a regular (non-proxy) route, you can just use `res.locals.user_session` that is exposed from `verifyRequest` to get user's online session tokens.


I'm guessing you want to make calls to Shopify to fetch orders / products / something else, and you're better off making an Express route and pulling in that way. Use `clientProvider` to make requests instead of having to worry about doing it manually. Also, since you're using this as a starter, please spend some time going through every middleware and util function that is exposed and documented with JSDocs so you can understand what you're building your entire app on^

**Reply by john-kuo** [reply]
got it.  thanks
